src/containers/TransactionPage/HostInformationModal/HostInformationModal.js

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { FormattedMessage, intlShape, injectIntl } from '../../../util/reactIntl';
+import { propTypes } from '../../../util/types';
+import { Modal } from '../../../components';
+
+import css from './HostInformationModal.module.css';
+import HostInformationForm from '../HostInformationForm/HostInformationForm';
+
+const HostInformationModal = props => {
+  const {
+    className,
+    rootClassName,
+    id,
+    marketplaceName,
+    intl,
+    isOpen,
+    onCloseModal,
+    onManageDisableScrolling,
+    onSubmitHostInfo,
+    hostInfoSent,
+    sendHostInfoInProgress,
+    sendHostInfoError,
+    listing,
+    listingAnimals,
+    onFetchAnimalListing,
+  } = props;
+
+  const classes = classNames(rootClassName || css.root, className);
+  const closeButtonMessage = intl.formatMessage({ id: 'HostInfoModal.later' });
+
+  useEffect(() => {
+    onFetchAnimalListing(listing.id?.uuid)
+  }, [listing.id?.uuid])
```

Comment:
Why is this not in the duck file?

Reply:
I mapped fetchAnimalListingByACC thunk to onFetchAnimalListing in the file TransactionPage.js. Then I passed it down to the HostInformationModal.js. I intended to fetch the animal listing based on the listing ID, so I placed it in the HostInformationModal and it only fetches when the listing.id?.uuid changes.